crates/core/thin_edge_json/src/event.rs

```diff
@@ -39,27 +42,47 @@ pub mod error {
 
         #[error(transparent)]
         SerdeJsonError(#[from] serde_json::error::Error),
+
+        #[error(transparent)]
+        JsonWriterError(#[from] JsonWriterError),
     }
 }
 
 impl ThinEdgeEvent {
     pub fn try_from(
         mqtt_topic: &str,
         mqtt_payload: &str,
+        may_be_child: Option<&str>,
     ) -> Result<Self, ThinEdgeJsonDeserializerError> {
         let topic_split: Vec<&str> = mqtt_topic.split('/').collect();
-        if topic_split.len() == 3 {
+        if topic_split.len() == 3 || topic_split.len() == 4 {
```

Comment:
I feel this function's error handling is still weak.

It accepts these cases, but both must be errors:

- `topic_split.len() == 3` AND `may_be_child == Some("child-id")`
- `topic_split.len() == 4` AND `may_be_child == None`

Reply:
And I think these two cases are candidates of unit tests.

Reply:
Now I derive the child id inside using the topic.